main.rb

```diff
@@ -0,0 +1,158 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    arr = to_a if self.class == Hash || self.class == Range
+    while i < size
+      if self.class == Hash || self.class == Range
+        yield(arr.at(i))
+      else
+        yield(at(i))
+      end
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index(*arg)
+    return to_enum(:my_each_with_index) unless block_given? && arg.empty?
+
+    index = 0
+    arr = self
+    tarr = arr.to_a
+    i = 0
+    while i < arr.size
+      yield(tarr[i], index)
+      index += 1
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    item = []
+    arr = self
+    tarr = []
+    if arr.class == Hash
+      arr.my_each { |key, _value| tarr << key }
+    else
+      tarr = arr.to_a
+    end
+    i = 0
+    while i < tarr.size
+      yield(tarr[i]) ? item << tarr[i] : nil
+      i += 1
+    end
+    item
+  end
+
+  def my_all?(*args)
+    flag = 0
+    i = 0
+    arr = self
+    is_a?(Range) || is_a?(Hash) ? arr = arr.to_a : nil
+
+    case args.empty?
+    when true
+      if block_given?
+        blk = yield(arr.at(i))
+        arr.my_each { flag += 1 if blk == true }
+      else
+        arr.my_each { flag += 1 unless i.nil? || i == false }
+      end
+    when false
+      arr.my_each { flag += 1 if i.is_a?(args[0]) }
+    end
+    return true if flag == size
+    return false if flag != size
+  end
```

Comment:
# my_all? 
1. [ ] When no block or argument is given returns false when one of the collection members is false or nil

     Failure/Error: expect(false_array.my_all?).to be false_array.all?

       expected false
            got true

##
# my_all?
2. [ ] When a class is passed as an argument returns false if any of the collection is not a member of such class

     Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)

       expected false
            got true
##

# my_all? 
3. [ ] When a Regex is passed as an argument returns true if all of the collection matches the Regex

     Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)

     TypeError:
       class or module required
##
# my_all? 
4. [ ] When a Regex is passed as an argument returns false if any of the collection does not match the Regex

     Failure/Error: expect(words.my_all?(/o/)).to be words.all?(/o/)

     TypeError:
       class or module required
##

# my_all? 
5. [ ] When a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

     Failure/Error: expect(array.my_all?(3)).to be array.all?(3)

     TypeError:
       class or module required
##     
# my_all? 
6. [ ] When a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern

     Failure/Error: expect(array.my_all?(3)).to be array.all?(3)

     TypeError:
       class or module required